server/core/src/main/java/org/infinispan/server/core/AbstractProtocolServer.java

```diff
@@ -100,6 +100,8 @@ public void start(C configuration, EmbeddedCacheManager cacheManager) {
          throw new IllegalStateException("CacheIgnoreManager is a required component");
       }
 
+      bcr.registerComponent(getQualifiedName(), this, false);
```

Comment:
I don't think using `getQualifiedName()` is ok, because the javadoc of `ProtocolServerConfigurationBuilder.name()` says

```
    * Specifies a custom name for this server in order to easily distinguish it from other servers, e.g. via JMX. Defaults to the empty string.
```

and at least some of our tests have interpreted that as "set a different name on each node".

Reply:
The test then needs to be fixed: the original jmx code uses getQualifiedName() as well.
The intention of `name()` is to give different endpoints of the same type on the same instance a unique name. I'll add a note specifying that it needs to be the same across the cluster.

Reply:
@danberindei are you referring to https://github.com/infinispan/infinispan/blob/a6e618b93879c0705d5f574e55cdb05aa2ec7f14/client/hotrod-client/src/test/java/org/infinispan/client/hotrod/xsite/AbstractHotRodSiteFailoverTest.java#L114 ?

Reply:
@tristantarrant it's not the only one, e.g.

https://github.com/infinispan/infinispan/blob/485e561a27058e3cef3b84084211ce24e87bb250/server/rest/src/test/java/org/infinispan/rest/resources/AbstractRestResourceTest.java#L138
https://github.com/infinispan/infinispan/blob/b05ee36732090ecbbaff4f851a3252d4fd5d96a8/server/rest/src/test/java/org/infinispan/rest/search/BaseRestSearchTest.java#L86
https://github.com/infinispan/infinispan/blob/aad7cba441e8bf4e2acdf8c870f9775df7b089bf/server/rest/src/test/java/org/infinispan/rest/resources/XSiteResourceTest.java#L75

Reply:
Oops, I didn't see an answer here and I assumed you didn't change the tests